homeassistant/components/rtsptowebrtc/__init__.py

```diff
@@ -0,0 +1,89 @@
+"""RTSPtoWebRTC integration with an external RTSPToWebRTC Server.
+
+WebRTC uses a direct communication from the client (e.g. a web browser) to a
+camera device. Home Assistant acts as the signal path for initial set up,
+passing through the client offer and returning a camera answer, then the client
+and camera communicate directly.
+
+However, not all cameras natively support WebRTC. This integration is a shim
+for camera devices that support RTSP streams only, relying on an external
+server RTSPToWebRTC that is a proxy. Home Assistant does not participate in
+the offer/answer SDP protocol, other than as a signal path pass through.
+
+Other integrations may use this integration with these steps:
+- Check if this integration is loaded
+- Call is_suported_stream_source for compatibility
+- Call async_offer_for_stream_source to get back an answer for a client offer
+"""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+import logging
+
+import async_timeout
+from rtsp_to_webrtc.client import Client
+from rtsp_to_webrtc.exceptions import ClientError, ResponseError
+
+from homeassistant.components import camera
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant
+from homeassistant.exceptions import ConfigEntryNotReady, HomeAssistantError
+from homeassistant.helpers.aiohttp_client import async_get_clientsession
+
+_LOGGER = logging.getLogger(__name__)
+
+DOMAIN = "rtsptowebrtc"
+DATA_SERVER_URL = "server_url"
+DATA_UNSUB = "unsub"
+TIMEOUT = 10
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up RTSPtoWebRTC from a config entry."""
+    hass.data.setdefault(DOMAIN, {})
+
+    if DATA_SERVER_URL not in entry.data:
```

Comment:
When can this happen?

Reply:
Should never happen as the error message states its invalid. I'm getting a hint that you'd prefer not to do this check, so removing.

Reply:
Yes. We assume we're in control of the config entry data.